src/realm/backup_restore.cpp

```diff
@@ -0,0 +1,172 @@
+/*************************************************************************
+ *
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **************************************************************************/
+
+#include <realm/backup_restore.hpp>
+#include <realm/util/file.hpp>
+
+#include <vector>
+#include <chrono>
+
+namespace realm {
+
+/*
+ * IMPORTANT: The following two arrays must be kept updated
+ * as new versions are released or if rollback is ever done.
+ */
+
+using version_list_t = BackupHandler::version_list_t;
+using version_time_list_t = BackupHandler::version_time_list_t;
+
+// Note: accepted versions should have new versions added at front
+const version_list_t accepted_versions_{20, 11, 10, 9, 8, 7, 6, 0};
+
+// the pair is <version, age-in-seconds>
+// we keep backup files in 3 months: 3*30*24*60*60 secs
+constexpr int three_months = 3 * 30 * 24 * 60 * 60;
```

Comment:
Maybe use 31 days so it is at least 3 calendar months? Doesn't really matter though.